akka-actor/src/main/scala/akka/pattern/Patterns.scala

```diff
@@ -336,9 +365,28 @@ object PatternsCS {
    * @param messageFactory function taking an actor ref and returning the message to be sent
    * @param timeout        the timeout for the response before failing the returned completion operator
    */
+  @deprecated("Use the overloaded one which accepts java.time.Duration instead.", since = "2.5.14")
   def askWithReplyTo(actor: ActorRef, messageFactory: japi.function.Function[ActorRef, Any], timeout: Timeout): CompletionStage[AnyRef] =
     extended.ask(actor, messageFactory.apply _)(timeout).toJava.asInstanceOf[CompletionStage[AnyRef]]
 
+  /**
+   * A variation of ask which allows to implement "replyTo" pattern by including
+   * sender reference in message.
+   *
+   * {{{
+   * final CompletionStage<Object> f = PatternsCS.askWithReplyTo(
+   *   worker,
+   *   askSender -> new Request(askSender),
+   *   timeout);
+   * }}}
+   *
+   * @param actor          the actor to be asked
+   * @param messageFactory function taking an actor ref and returning the message to be sent
+   * @param timeout        the timeout for the response before failing the returned completion stage
+   */
+  def askWithReplyTo(actor: ActorRef, messageFactory: japi.function.Function[ActorRef, Any], timeout: java.time.Duration): CompletionStage[AnyRef] =
```

Comment:
It is not possible right because of the Java generics system as i already mentioned above. We will still have the ambiguity.
See:
```
def askWithReplyTo(actor: ActorRef, messageFactory: japi.function.Function[ActorRef, Any], duration: java.time.Duration): CompletionStage[AnyRef] = ...

def askWithReplyTo(actor: ActorRef, messageFactory: java.util.function.Function[ActorRef, Any], duration: java.time.Duration): CompletionStage[AnyRef] = ...
```

At this point there is an ambiguity for the compile at the client side (akka.pattern.PatternTest.java).
``` japi.function.Function[ActorRef, Any]``` and ```java.util.function.Function[ActorRef, Any]```are represented the same in java ```java.util.function.Function<ActorRef, Object>```.